### ISSUE TITLE
Content bug: Template literals erroneously called "strings", claims that spec used to do so

### ISSUE THREAD

**Issue by tjcrowder** [mdn/content#7942]
## What page(s) did you find the problem on?

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

## Specific page section or heading?

The two introductory paragraphs.

## What is the problem?

The first two paragraphs are currently:

> Template literals are string literals allowing embedded expressions. You can use multi-line strings and string interpolation features with them.
>
> They were called "template strings" in prior editions of the ES2015 specification.

Problems:

1. *"...are string literals..."* Template literals aren't string literals. If they were string literals, you could use them anywhere a string literal can be used, but you can't. For example, you can't use them for static import module specifiers, non-computed property keys in object literals, or `use strict`.

2. *"They were called "template strings" in prior editions..."* There's no edition of the specification I know of that called them "template strings," I'd be curious to know what that author was referring to. They were added in [ES2015](https://262.ecma-international.org/6.0/), which reliably calls them template literals. ("TemplateStrings" does occur in the text, but it's specifically in reference to getting the string portions of the template.) I wouldn't be surprised if they were called "template strings" in draft versions and/or proposals, but even if so, noting that here just perpetuates the misconception that these are a third kind of string literal.

## What did you expect to see?

I think it should be clear that they're their own kind of literal, not string literals, and the claim about the spec should just be removed in favor of just flagging up the confusion/informal usage.

I'd reword the first paragraph and replace the second.

> Template literals are literals delimited with backticks (<code>`</code>) allowing embedded expressions called *substitutions*. *Untagged* template literals result in strings, which is useful for string interpolation (and multiline strings, since unescaped newlines are allowed). *Tagged* template literals call a function (the *tag function*) with an array of any text segments from the literal followed by arguments with the values of any substitutions, which is useful for [DSLs](https://en.wikipedia.org/wiki/Domain-specific_language).
>
> Template literals are sometimes informally called *template strings*, but they aren't string literals and can't be used everywhere a string literal can be used. Also, a tagged template literal may not result in a string; it's up to the tag function what it creates (if anything).

Happy to do a PR if it suits.

I think I'd reorganize the page a bit as well (in fact, for me untagged template literals and tagged template literals are each big enough that two subpages might be justified). I could take that on in consultation with the team if you'd like (though not for 3-4 weeks). (FWIW, I'm a [published author](https://www.wiley.com/en-gb/JavaScript:+The+New+Toys-p-9781119367963) in this area.)

## Did you test this? If so, how?

N/A

**Comment by sideshowbarker** [comment]
> Happy to do a PR if it suits.

Your rewrite sounds perfect to me, so I’d be happy to review a PR with those changes.

> I think I'd reorganize the page a bit as well (in fact, for me untagged template literals and tagged template literals are each big enough that two subpages might be justified). I could take that on in consultation with the team if you'd like

I believe those further changes would be very much welcome, and I personally be very happy to help with any feedback and review you’d need.

> (though not for 3-4 weeks)

I think there’s no rush — and you’d not even need to open an issue for tracking; you could just work on putting a patch together and then raise a PR for it when you have something ready for feedback and review (and we could iterate on it as needed, there in that PR branch).

**Comment by tjcrowder** [comment]
@sideshowbarker Thanks! I'll put a note in my calendar to revisit the page when things calm down a bit. Meanwhile, I've done the PR. Thanks again!
